torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1803,6 +1824,18 @@ def _make_data_dependent_error(self, expr, unhinted_expr):
             # problem
         )
 
+    def _set_replacement(self, a: "sympy.Symbol", expr: "sympy.Expr") -> None:
+        """
+        Adds or updates a replacement for a symbol.
+        Use this instead of `self.replacements[a] = expr`.
+        """
+        if a not in self.replacements or expr != self.replacements[a]:
+            if torch._dynamo.config.print_specializations and isinstance(expr, (sympy.Integer, sympy.Float)):
+                # specializing to a constant, which is likely unexpected
+                torch._dynamo.guards.log.warning(f"Specializing {self.var_to_sources[a][0].name()} to {expr}")
+                torch._dynamo.guards.log.debug(f"Stack trace:\n{get_debugging_stack()}")
```

Comment:
Instead of doing this by hand, you can just call `stack_info=True` to the logging call

Reply:
done